test/unit_tests/arkode/CXX_serial/ark_test_getjac.cpp

```diff
@@ -0,0 +1,273 @@
+  if (check_ptr(A, "SUNDenseMatrix")) return 1;
+
+  SUNLinearSolver LS = SUNLinSol_Dense(y, A, sunctx);
+  if (check_ptr(LS, "SUNLinSol_Dense")) return 1;
+
+  flag = ARKStepSetLinearSolver(arkode_mem, LS, A);
+  if (check_flag(flag, "ARKStepSetLinearSolver")) return 1;
+
+  sunrealtype udata[4] = {-TWO, HALF, HALF, -ONE};
+
+  flag = ARKStepSetUserData(arkode_mem, udata);
+  if (check_flag(flag, "ARKStepSetUserData")) return 1;
+
+  // Initial time and fist output time
+  sunrealtype tret = ZERO;
+  sunrealtype tout = tret + SUN_RCONST(0.1);
+
+  // Advance one step in time
+  flag = ARKStepEvolve(arkode_mem, tout, y, &tret, ARK_ONE_STEP);
+  if (check_flag(flag, "ARKStep")) return 1;
+
+  // Get the internal finite difference approximation to J
+  SUNMatrix Jdq;
+  flag = ARKStepGetJac(arkode_mem, &Jdq);
+  if (check_flag(flag, "ARKStepGetJac")) return 1;
+
+  // Get the step and time at which the approximation was computed
+  long int nst_Jdq;
+  flag = ARKStepGetJacNumSteps(arkode_mem, &nst_Jdq);
+  if (check_flag(flag, "ARKStepGetJacNumSteps")) return 1;
+
+  sunrealtype t_Jdq;
+  flag = ARKStepGetJacTime(arkode_mem, &t_Jdq);
+  if (check_flag(flag, "ARKStepGetJacTime")) return 1;
+
+  // Compute the true Jacobian
+  SUNMatrix Jtrue = SUNDenseMatrix(2, 2, sunctx);
+  if (check_ptr(Jtrue, "SUNDenseMatrix")) return 1;
+
+  flag = ytrue(t_Jdq, y);
+  if (check_flag(flag, "ytrue")) return 1;
+
+  flag = J(t_Jdq, y, nullptr, Jtrue, &udata, nullptr, nullptr, nullptr);
+  if (check_flag(flag, "J")) return 1;
+
+  // Compare finite difference and true Jacobian
+  sunrealtype* Jdq_data = SUNDenseMatrix_Data(Jdq);
+  if (check_ptr(Jdq_data, "SUNDenseMatrix_Data")) return 1;
+
+  sunrealtype* Jtrue_data = SUNDenseMatrix_Data(Jtrue);
+  if (check_ptr(Jtrue_data, "SUNDenseMatrix_Data")) return 1;
+
+  // Output Jacobian data
+  std::cout << std::scientific;
+  std::cout << std::setprecision(std::numeric_limits<sunrealtype>::digits10);
+  std::cout << "Jac nst = " << nst_Jdq << std::endl;
+  std::cout << "Jac t   = " << t_Jdq << std::endl;
+  std::cout << std::endl;
+  std::cout << std::setw(8) << std::right << "Index" << std::setw(25)
+            << std::right << "J DQ" << std::setw(25) << std::right << "J true"
+            << std::setw(25) << std::right << "absolute difference"
+            << std::setw(25) << std::right << "relative difference" << std::endl;
+  for (int i = 0; i < 4 * 25 + 8; i++) std::cout << "-";
+  std::cout << std::endl;
+
+  sunindextype ldata = SUNDenseMatrix_LData(Jtrue);
+  for (sunindextype i = 0; i < ldata; i++)
+  {
+    std::cout << std::setw(8) << std::right << i << std::setw(25) << std::right
+              << Jdq_data[i] << std::setw(25) << std::right << Jtrue_data[i]
+              << std::setw(25) << std::right
+              << std::abs(Jdq_data[i] - Jtrue_data[i])
+              << std::setw(25) << std::right
+              << std::abs(Jdq_data[i] - Jtrue_data[i])/Jtrue_data[i]
+              << std::endl;
+  }
+
```

Comment:
In addition to writing the Jacobian matrices to the screen, should we compute some norm of the difference and verify that it is "small enough"?  For example, I would think that these should have entries with relative error smaller  than 100*sqrt(unit_roundoff).